src/overlays/actors/ovl_En_Rl/z_en_rl.c

```diff
@@ -9,81 +16,379 @@ void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx);
+}
+
+void func_80AE7D40(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE744C(this, globalCtx);
+    func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE7AF8(this, globalCtx);
+    func_80AE73D8(this, globalCtx);
+}
+
+void func_80AE7D94(EnRl* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s16 temp = this->unk_190;
+    s32 addr = D_80AE81A0[temp];
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_rl_inKenjyanomaDemo02.c", 304);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x08, SEGMENTED_TO_VIRTUAL(addr));
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x09, SEGMENTED_TO_VIRTUAL(addr));
+    gDPSetEnvColor(oGfxCtx->polyXlu.p++, 0, 0, 0, this->alpha);
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x0C, &D_80116280);
+
+    oGfxCtx->polyXlu.p = SkelAnime_DrawSV2(globalCtx, skelAnime->skeleton, skelAnime->limbDrawTbl,
+                                           skelAnime->dListCount, NULL, NULL, NULL, oGfxCtx->polyXlu.p);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_rl_inKenjyanomaDemo02.c", 331);
+}
+
+void EnRl_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnRl* this = THIS;
+
+    if ((this->action < 0) || (this->action > 7) || (sActionFuncs[this->action] == 0)) {
+        osSyncPrintf(VT_FGCOL(RED) "メインモードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
+    } else {
+        sActionFuncs[this->action](this, globalCtx);
+    }
+}
+
+void EnRl_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnRl* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 50.0f);
+    if (this->actor.params == 2) {
+        func_80AE7878(this, globalCtx);
+    } else {
+        func_80AE7544(this, globalCtx);
+    }
+}
+void func_80AE7FD0(EnRl* this, GlobalContext* globalCtx) {
+}
+
+void func_80AE7FDC(EnRl* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s16 temp = this->unk_190;
+    s32 addr = D_80AE81A0[temp];
+    SkelAnime* skelAnime = &this->skelAnime;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_rl.c", 416);
+
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPSegment(oGfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(addr));
+    gSPSegment(oGfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(addr));
+    gDPSetEnvColor(oGfxCtx->polyOpa.p++, 0, 0, 0, 255);
+    gSPSegment(oGfxCtx->polyOpa.p++, 0x0C, &D_80116280[2]);
+
+    SkelAnime_DrawSV(globalCtx, skelAnime->skeleton, skelAnime->limbDrawTbl, skelAnime->dListCount, NULL, NULL,
+                     &this->actor);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_rl.c", 437);
+}
+
+void EnRl_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnRl* this = THIS;
+
+    if (this->drawConfig < 0 || this->drawConfig >= 3 || sDrawFuncs[this->drawConfig] == NULL) {
+        osSyncPrintf(VT_FGCOL(RED) "描画モードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
+        return;
+    }
+    sDrawFuncs[this->drawConfig](this, globalCtx);
```

Comment:
`else` instead of an early return?

Reply:
early return makes sense for cases where its error checking. if we have an early actor kill or early return cause of an oob check or something, we do generally use early return 
its really situational and one isnt really more correct over the other, but in this case i think its fine

Reply:
Sure but this should at least be consistent in a single file, see `EnRl_Update`

Reply:
yeah i agree, update should be early return as well